FHEM/00_SIGNALduino.pm

```diff
@@ -1249,7 +1250,8 @@ sub SIGNALduino_CheckVersionResp
 		RemoveInternalTimer($hash);
 	}
 
-	$msg =~ m/($gets{$hash->{getcmd}->{cmd}}[2])/;
+	### ToDo, manchmal kommen Mu Nachrichten in $msg und somit ist keine Version feststellbar !!!
```

Comment:
Da müsste dann ja der Zeilenumbruch fehlen, das ist ja sehr seltsam. Früher ist uns das nicht passiert.

Reply:
Mir fiel es nur auf wenn ich restarte und mal geht’s und mal nicht. Bei der Hilfsausgabe sah ich den Fehler.

Reply:
Ich konnte nun verifizieren wieso es dazu kommt. Sobald beim Restart bzw. Init Sensordaten eintreffen, dan geht dies schief. Sobald keine Daten eintreffen zu dem Zeitpunkt, geht es gut. Das kann ich hier shr gut in dem Sensordschungel nachvollziehen.

```
Beispiel für einen Erfolg - Init!
--------------------------------
2020.01.04 11:04:19 0: Featurelevel: 5.9
2020.01.04 11:04:20 1: radino_433Mhz: SIGNALduino_SimpleWrite_XQ
2020.01.04 11:04:20 3: radino_433Mhz: SimpleWrite_XQ, disable receiver (XQ)
2020.01.04 11:04:20 3: radino_433Mhz: StartInit, get version, retry = 0
2020.01.04 11:04:20 1: radino_433Mhz: Read, msg: V 3.4.0-dev SIGNALduino cc1101 (433 Mhz ) - compiled at Dec  4 2019 21:59:32
2020.01.04 11:04:20 1: radino_433Mhz: Read, msg: regexp=V\s.*SIGNAL(duino|ESP).* cmd=version msg=V 3.4.0-dev SIGNALduino cc1101 (433 Mhz ) - compiled at Dec  4 2019 21:59:32
2020.01.04 11:04:20 1: radino_433Mhz: SIGNALduino_CheckVersionResp - msg:V 3.4.0-dev SIGNALduino cc1101 (433 Mhz ) - compiled at Dec  4 2019 21:59:32
2020.01.04 11:04:20 2: radino_433Mhz: CheckVersionResp, initialized v3.4.2_dev_02.01
2020.01.04 11:04:20 3: radino_433Mhz: CheckVersionResp, enable receiver (XE) 
2020.01.04 11:04:29 1: radino_433Mhz: Read, msg: MC;LL=-2714;LH=3149;SL=-1253;SH=1669;D=2B65FFDE;C=1464;L=32;R=245;

Beispiel für einen Fehlschlag!
---------------------------------------
2020.01.04 11:04:19 0: Featurelevel: 5.9
2020.01.04 11:04:20 1: nano_433Mhz: SIGNALduino_SimpleWrite_XQ
2020.01.04 11:04:20 3: nano_433Mhz: SimpleWrite_XQ, disable receiver (XQ)
2020.01.04 11:04:20 3: nano_433Mhz: StartInit, get version, retry = 0
2020.01.04 11:04:20 1: nano_433Mhz: Read, msg: MC;LL=-1014;LH=929;SL=-527;SH=455;D=51EE68B5978C178EAB9E818;C=4Mu;��;�ā;���;���;���;d!!!!!!!!!4!!4!4444!4!!4!44!!!4!4!!4!44!4!!!4444!!44!4!4!!!!4!!!44!4!!4!44!44!440;C3;R57;
2020.01.04 11:04:20 1: nano_433Mhz: Read, msg: regexp=V\s.*SIGNAL(duino|ESP).* cmd=version msg=MC;LL=-1014;LH=929;SL=-527;SH=455;D=51EE68B5978C178EAB9E818;C=4Mu;��;�ā;���;���;���;d!!!!!!!!!4!!4!4444!4!!4!44!!!4!4!!4!44!4!!!4444!!44!4!4!!!!4!!!44!4!!4!44!44!440;C3;R57;
2020.01.04 11:04:20 1: nano_433Mhz: Read, msg: MC;LL=-1008;LH=939;SL=-527;SH=446;D=51EE68B5978C178EAB9E818;C=486;L=89;R=63;
2020.01.04 11:04:20 1: nano_433Mhz: Read, msg: MC;LL=-1006;LH=948;SL=-526;SH=445;D=F7345AEBC60BC755CE57C;C=487;L=82;R=61;
2020.01.04 11:04:20 1: nano_433Mhz: Read, msg: MC;LL=-1006;LH=948;SL=-529;SH=450;D=A8F7345ADBC60BC755CFCDC;C=488;L=90;R=62;
2020.01.04 11:04:20 1: nano_433Mhz: Read, msg: V 3.4.0-dev SIGNALduino cc1101 (chip CC1101) - compiled at Dec  4 2019 22:02:15
2020.01.04 11:04:30 1: nano_433Mhz: SIGNALduino_CheckVersionResp - msg:
2020.01.04 11:04:30 1: nano_433Mhz: CheckCmdResp, Not an SIGNALduino device, got for V:  
2020.01.04 11:04:30 2: nano_433Mhz: CloseDevice, closed
```

Reply:
Ah, vermutlich wäre es am besten, den Empfangspuffer bei einem restart zu leeren :)

Reply:
Ich hab da was eingebaut, was dieses Problem beheben sollte

Reply:
Ich bin gespannt :-)
Ich werde es verifizieren.

Reply:
Zu früh gefreut. eingespielt und RESTART und erneut error. Ging nicht.